### PR TITLE
Breaking: deprecate react-icheck

### PR DESCRIPTION
**Pull request by lightbringer1991**
- Remove `react-icheck` from dependency list.
- Remove `react-icheck` related css styles
- Introduce hover state for `Checkbox` component (since the svg is already there)
after

TODO:

- [ ] check with related projects

## Description

`react-icheck` is no longer needed.
The SVGs are actually ours, there is no style coming from `react-icheck`.

## Motivation and Background Context

Resolves https://github.com/Adslot/adslot-ui/issues/727

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

## Check-list:

- [x] I have read the [Contributing](CONTRIBUTING.md) document.
- [ ] I've thought about and labelled my PR/commit message appropriately.
- [ ] If this PR introduces breaking changes I've described the impact and migration path for existing applications.
- [ ] CI is green (coverage, linting, tests).
- [ ] I have updated the documentation accordingly.
- [ ] I've two LGTMs/Approvals.
- [ ] I've fixed or replied to all my code-review comments.
- [ ] I've manually tested with a buddy.
- [ ] I've squashed my commits into one.

### CODE REVIEW

**Comment by ChaoLiangSuper** [comment]
Just realized one thing. Do we need to change the iCheck class name (the class name for the SVG) for some else class name in this PR?

**Comment by omgaz** [comment]
screen shot 2018-06-07 at 09 55 30
screen shot 2018-06-07 at 09 55 40
screen shot 2018-06-07 at 09 56 04


Just some renames to clean-up and move the icheck custom stuff to the radio/checkbox scss stuff

**Comment by lightbringer1991** [comment]
yea that would turn this PR into a breaking change. I'll do that then.

File: src/components/adslot-ui/Checkbox/index.jsx [hunk 1]
```diff
@@ -37,11 +51,21 @@ class Checkbox extends React.Component {
...
       id,
       className,
+      onMouseEnter: this.onMouseEnter,
+      onMouseLeave: this.onMouseLeave,
     };
 
+    const iconClassNames = _([
+      'selection-component-icon',
+      this.state.checked ? 'checked' : null,
+      this.state.isHover ? 'hover' : null,
```
Comment (mdotwills) [inline src/components/adslot-ui/Checkbox/index.jsx]: _suggestion:_ if we want to reduce our dependency on lodash we could

```javascript
const iconClassNames = ['selection-component-icon'];
if (this.state.checked) iconClassNames.push('checked');
if (this.state.isHover) iconClassNames.push('hover');
```

**Comment by lightbringer1991** [comment]
@mdotwills this is already a breaking change PR, combining with the previous merged PR (which is also a breaking change), we're looking at releasing a major version of `adslot-ui`

File: src/components/adslot-ui/Checkbox/index.jsx (hunk 1 above)
Comment (omgaz) [reply src/components/adslot-ui/Checkbox/index.jsx]: We already have classNames library that's also part of React that does this:

```es6
import cx from 'classNames';

const iconClassNames = cx([
  'selection-component-icon',
  {
    'checked': this.state.checked,
    'hover': this.state.isHover,
  },
]
```

File: src/styles/mixins/enhancedSwitch.scss [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+@mixin enhanced-switch-icon {
+  background-repeat: no-repeat;
+  border: 0;
+  cursor: pointer;
+  display: inline-block;
+  height: 16px;
+  position: relative;
+  vertical-align: middle;
+  width: 16px;
```
Comment (omgaz) [inline src/styles/mixins/enhancedSwitch.scss]: Do you think it makes sense to use `$svg-icon-size` for width/height or do you feel we want to keep them separate?

Comment (omgaz) [reply src/styles/mixins/enhancedSwitch.scss]: Can you also keep width & height together?

Comment (lightbringer1991) [reply src/styles/mixins/enhancedSwitch.scss]: I didn't know we already have a variable, will switch over

**Comment by lightbringer1991** [comment]
stale, closing

**Pull request closed without merging**